src/main/python/deep_qa/training/trainer.py

```diff
@@ -42,6 +42,27 @@ def __init__(self, params: Dict[str, Any]):
             parent_directory = os.path.dirname(self.model_prefix)
             os.makedirs(parent_directory, exist_ok=True)
 
+        # Preferred backend to use for training. If a different backend is detected, we still
+        # train but we also warn the user.
+        self.preferred_backend = params.pop('preferred_backend', None)
+        if self.preferred_backend and self.preferred_backend.lower() != K.backend():
+            warning_info = ("@ Preferred backend is %s, but "
```

Comment:
One last nitpick - this is a lot of logic to be sitting in `__init__`.  Can you move it to a function like `_print_backend_warning()`?